src/Entity/EntityMeta.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\entity_meta_relation\Entity;
+
+use Drupal\Core\Entity\EntityChangedTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\RevisionableContentEntityBase;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\entity_meta_relation\EntityMetaInterface;
+
+/**
+ * Defines the entity meta entity class.
+ *
+ * @ContentEntityType(
+ *   id = "entity_meta",
+ *   label = @Translation("Entity meta"),
+ *   label_collection = @Translation("Entity metas"),
+ *   bundle_label = @Translation("Entity meta type"),
+ *   handlers = {
+ *     "view_builder" = "Drupal\entity_meta_relation\EntityMetaViewBuilder",
+ *     "list_builder" = "Drupal\entity_meta_relation\EntityMetaListBuilder",
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *     "form" = {
+ *       "add" = "Drupal\entity_meta_relation\Form\EntityMetaForm",
+ *       "edit" = "Drupal\entity_meta_relation\Form\EntityMetaForm",
+ *       "delete" = "Drupal\Core\Entity\ContentEntityDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *     }
+ *   },
+ *   base_table = "entity_meta",
+ *   revision_table = "entity_meta_revision",
+ *   show_revision_ui = TRUE,
+ *   admin_permission = "administer entity meta types",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "revision" = "revision_id",
+ *     "bundle" = "bundle",
+ *     "label" = "id",
+ *     "uuid" = "uuid"
+ *   },
+ *   revision_metadata_keys = {
+ *     "revision_created" = "revision_timestamp",
+ *     "revision_log_message" = "revision_log"
+ *   },
+ *   links = {
+ *     "add-form" = "/admin/content/entity-meta/add/{entity_meta_type}",
+ *     "add-page" = "/admin/content/entity-meta/add",
+ *     "canonical" = "/entity_meta/{entity_meta}",
+ *     "edit-form" = "/admin/content/entity-meta/{entity_meta}/edit",
+ *     "delete-form" = "/admin/content/entity-meta/{entity_meta}/delete",
+ *     "collection" = "/admin/content/entity-meta"
+ *   },
+ *   bundle_entity_type = "entity_meta_type",
+ *   field_ui_base_route = "entity.entity_meta_type.edit_form"
+ * )
+ */
+class EntityMeta extends RevisionableContentEntityBase implements EntityMetaInterface {
```

Comment:
Shouldn't we from the start make the entity type translatable (and all relevant fields) so that if later we need to support meta information different per language on the same entity, we can use that?